src/libs/ReportUtils.ts

```diff
@@ -6851,6 +6852,10 @@ function isNonAdminOrOwnerOfPolicyExpenseChat(report: OnyxInputOrEntry<Report>,
  * Whether the user can join a report
  */
 function canJoinChat(report: OnyxInputOrEntry<Report>, parentReportAction: OnyxInputOrEntry<ReportAction>, policy: OnyxInputOrEntry<Policy>): boolean {
+    if (isPublicRoom(report as OnyxEntry<Report>) && SessionUtils.isAnonymousUser()) {
```

Comment:
I don't like this typecast. Any other way? There's no such instance in this file.

Reply:
Actually, why `isPublicRoom` condition needed here? Isn't `isAnonymousUser` check enough?

Reply:
> Actually, why isPublicRoom condition needed here? Isn't isAnonymousUser check enough?

If user has already signed in and joined the public room, we do not need to display the "Join" button.

Reply:
> > Actually, why isPublicRoom condition needed here? Isn't isAnonymousUser check enough?
> 
> If user has already signed in and joined the public room, we do not need to display the "Join" button.

I think you misunderstood my question. I asked about the necessity of isPublicRoom, not isAnonymousUser.
If isAnonymousUser, it's already public room, isn't it?